src/components/MeasureRoutes/MeasureRoutes.test.tsx

```diff
@@ -0,0 +1,153 @@
+import "@testing-library/jest-dom";
+// NOTE: jest-dom adds handy assertions to Jest and is recommended, but not required
+
+import * as React from "react";
+import { render } from "@testing-library/react";
+import MeasureBrowserRoutes, { MeasureRoutes } from "./MeasureRoutes";
+import { MemoryRouter } from "react-router";
+import { ApiContextProvider, ServiceConfig } from "../../api/ServiceContext";
+import { describe, expect, test } from "@jest/globals";
+import useMeasureServiceApi, {
+  MeasureServiceApi,
+} from "../../api/useMeasureServiceApi";
+import Measure from "../../models/Measure";
+import { act } from "react-dom/test-utils";
+
+jest.mock("../../api/useMeasureServiceApi");
+
+const useMeasureServiceApiMock =
+  useMeasureServiceApi as jest.Mock<MeasureServiceApi>;
+
+const MEASURE_CREATEDBY = "testuser@example.com";
+const measure = {
+  id: "measure ID",
+  createdBy: MEASURE_CREATEDBY,
+} as Measure;
+
+const serviceApiMock = {
+  fetchMeasure: jest.fn().mockResolvedValue(measure),
+  fetchMeasures: jest.fn().mockResolvedValue({
+    content: [measure],
+    totalPages: 1,
+    totalElements: 1,
+    numberOfElements: 1,
+    pageable: {
+      sort: {
+        empty: false,
+        sorted: true,
+        unsorted: false,
+      },
+      offset: 0,
+      pageNumber: 0,
+      pageSize: 10,
+      paged: true,
+      unpaged: false,
+    },
+  }),
+} as unknown as MeasureServiceApi;
+
+useMeasureServiceApiMock.mockImplementation(() => {
+  return serviceApiMock;
+});
+
+jest.mock("@madie/madie-util", () => ({
+  useOktaTokens: () => ({
+    getAccessToken: () => "test.jwt",
+    getUserName: () => "TestUser@example.com", //#nosec
+  }),
+  measureStore: {
+    state: null,
+    initialState: null,
+    updateMeasure: (measure) => measure,
+  },
+}));
+
+const serviceConfig: ServiceConfig = {
+  measureService: {
+    baseUrl: "example-service-url",
+  },
+  elmTranslationService: {
+    baseUrl: "test-elm-service",
+  },
+};
+
+jest.mock("../notfound/NotFound", () => () => {
+  return (
+    <div data-testid="notfound-component-mocked">404 NotFound Component</div>
+  );
+});
+
+// react no op error is caused by two awaits in one it call. ignorable.
+describe("Measure Router", () => {
+  test("Render a default notFound page", async () => {
+    await act(async () => {
+      const { findByTestId } = await render(
```

Comment:
Might be different with async, but, in general, don't need to wrap render() with act() as its internals are already wrapped in act().

Locally, I removed the act() wrapper and the await on render() and didn't see any negative impact to the test execution.